src/librustc/ty/maps.rs

```diff
@@ -929,6 +963,14 @@ define_maps! { <'tcx>
     [] needs_drop_raw: needs_drop_dep_node(ty::ParamEnvAnd<'tcx, Ty<'tcx>>) -> bool,
     [] layout_raw: layout_dep_node(ty::ParamEnvAnd<'tcx, Ty<'tcx>>)
                                   -> Result<&'tcx Layout, LayoutError<'tcx>>,
+
+    [] dylib_dependency_formats: MetaDataByCrateNum(CrateNum)
+                                    -> Rc<Vec<(CrateNum, LinkagePreference)>>,
+
+    [] is_allocator: MetaDataByCrateNum(CrateNum) -> bool,
+    [] is_panic_runtime: MetaDataByCrateNum(CrateNum) -> bool,
+
+    [] extern_crate: MetaDataByCrateNum(CrateNum) -> Rc<Option<ExternCrate>>,
```

Comment:
Some of these are literally the same as querying the attributes of the crate root of some extern crate (they should also probably be replaced with an actual attribute query but that's another question), so to me they make more sense as a property of that `DefId` instead of that of the `CrateNum` (the latter would make sense for e.g. all `impl` items in a crate).